nifi-nar-bundles/nifi-geohash-bundle/nifi-geohash-processors/src/main/java/org/apache/nifi/processors/geohash/GeohashRecord.java

```diff
@@ -0,0 +1,449 @@
+            final String rawLongitudePath = context.getProperty(LONGITUDE_RECORD_PATH).evaluateAttributeExpressions(input).getValue();
+            RecordPath longitudePath = cache.getCompiled(rawLongitudePath);
+            final String rawgeohashPath = context.getProperty(GEOHASH_RECORD_PATH).evaluateAttributeExpressions(input).getValue();
+            RecordPath geohashPath = cache.getCompiled(rawgeohashPath);
+
+            while ((record = reader.nextRecord()) != null) {
+                boolean updated = false;
+                if (encode) {
+                    Object encodedGeohash = getEncodedGeohash(latitudePath, longitudePath, record, format, level);
+                    updated = updateRecord(GEOHASH_RECORD_PATH, encodedGeohash, record, paths);
+                } else {
+                    WGS84Point decodedPoint = getDecodedPointFromGeohash(geohashPath, record, format);
+                    if (decodedPoint != null) {
+                        updated = updateRecord(LATITUDE_RECORD_PATH, String.valueOf(decodedPoint.getLatitude()), record, paths)
+                                && updateRecord(LONGITUDE_RECORD_PATH, String.valueOf(decodedPoint.getLongitude()), record, paths);
+                    }
+                }
+
+                if (routingStrategy == RoutingStrategy.REQUIRE_ALL_ENRICHED) {
+                    if (!updated) { //If the routing strategy is Require_All_Enriched and there exists a record that is not updated, the entire flowfile should be route to REL_FAILURE
+                        targetRelationship = REL_FAILURE;
+                    }
+                } else {
+                    if (updated) {
+                        //If the routing strategy is Skip_Unenriched and there exists a record that is updated, the entire flowfile should be route to REL_SUCCESS
+                        //If the routing strategy is Split_Enriched_Unenriched and there exists a record that is updated, this record should be route to REL_SUCCESS
+                        targetRelationship = REL_SUCCESS;
+                    }
+                }
+
+                if (routingStrategy != RoutingStrategy.SPLIT || updated) {
+                    writer.write(record);
+                    foundCount++;
+                } else { //if the routing strategy is Split_Enriched_Unenriched and the record is not updated
+                    notFoundWriter.write(record);
+                    notFoundCount++;
+                }
+            }
+
+            final WriteResult writeResult = writer.finishRecordSet();
+            writer.close();
+
+            WriteResult notFoundWriterResult = null;
+
+            if (notFoundWriter != null) {
+                notFoundWriterResult = notFoundWriter.finishRecordSet();
+                notFoundWriter.close();
+            }
+
+            is.close();
+            os.close();
+
+            if (osNotFound != null) {
+                osNotFound.close();
+            }
```

Comment:
Can those `close()` method calls be removed since the streams are declared in the enclosing `try` block and should be closed automatically?

Reply:
Thanks for the comment! I have removed all the close() except the ones for `writer` and `notFoundWriter` because they need to be closed manually.